src/rpsCommandLine.js

```diff
@@ -0,0 +1,183 @@
+        return `\n${this.focused.content}`;
+      },
+      result() {
+        return {
+          questionCount: this.focused.questionCount,
+          displayTime: this.focused.displayTime,
+        };
+      },
+    });
+    return response.level;
+  }
+
+  async selectRule() {
+    const ruleIndex = Math.floor(Math.random() * 3);
+    process.stdout.write(
+      `CPUに対して${RpsCommandLine.DISPLAY_RULES[ruleIndex]}手を選んでください`,
+    );
+
+    await this.#wait(2000);
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+    return ruleIndex;
+  }
+
+  async selectCpuRps(questionCount, displayTime) {
+    const RPS = [
+      RpsCommandLine.ROCK,
+      RpsCommandLine.PAPER,
+      RpsCommandLine.SCISSORS,
+    ];
+    const cpuRpsSelections = [];
+    for (let i = 0; i < questionCount; i++) {
+      const rpsIndex = Math.floor(Math.random() * 3);
+      cpuRpsSelections[i] = RPS[rpsIndex];
+    }
+    for (let i = 0; i < questionCount; i++) {
+      process.stdout.write(`${i + 1}回目 CPU: ${cpuRpsSelections[i]}`);
+      await this.#wait(displayTime);
+      console.clear();
+    }
+    return cpuRpsSelections;
+  }
+
+  async #wait(ms) {
+    return new Promise((resolve) => setTimeout(resolve, ms));
+  }
+
+  async selectUserRps(questionCount) {
+    const userSelections = [];
+    for (let i = 0; i < questionCount; i++) {
+      const response = await enquirer.prompt({
+        type: "select",
+        name: "rps",
+        message: `${i + 1}回目の手を選んでください。`,
+        choices: [
+          { title: RpsCommandLine.ROCK },
+          { title: RpsCommandLine.PAPER },
+          { title: RpsCommandLine.SCISSORS },
+        ],
+      });
+      userSelections[i] = response.rps;
+    }
+    return userSelections;
+  }
+
+  showResult(questionCount, currentRule, cpuSelections, userSelections) {
+    const results = this.#judgeResults(
+      questionCount,
+      currentRule,
+      cpuSelections,
+      userSelections,
+    );
+    if (results.includes(false)) {
+      console.log(
+        `失敗！\n今回のルールは${RpsCommandLine.DISPLAY_RULES[currentRule]}手を選ぶことです。`,
+      );
+      for (let i = 0; i < questionCount; i++) {
+        if (results[i]) {
+          console.log(`${i + 1}回目：正解！`);
+        } else {
+          console.log(
+            `${i + 1}回目：不正解！CPUの選んだ手は${cpuSelections[i]}`,
+          );
+        }
+      }
+    } else {
+      console.log("成功！");
+    }
+  }
+
+  #judgeResults(questionCount, currentRule, cpuSelections, userSelections) {
+    const results = [];
+
+    for (let i = 0; i < questionCount; i++) {
+      switch (currentRule) {
+        case RpsCommandLine.WIN:
+          results.push(this.#winRule(userSelections[i], cpuSelections[i]));
+          break;
+        case RpsCommandLine.LOSE:
+          results.push(this.#loseRule(userSelections[i], cpuSelections[i]));
+          break;
+        default:
+          results.push(this.#drawRule(userSelections[i], cpuSelections[i]));
+          break;
+      }
```

Comment:
ruleの選択はループの手前でやった方が処理効率が良さそうです。
```suggestion
    const rule = switch ...;
    for (let i = 0; i < questionCount; i++) {
      results.push(rule(userSelections[i], cpuSelections[i]));
```

また、ループ処理はmapに置き換えるとコードを短くできそうです

```javascript
return cpuSelections.map(...
```

Reply:
mapに置き換えることで、`questionCount`が引数として不要となったため削除しました。